### PR TITLE
libs: Move aeabi_atexit from libxx/ to libc/machine/arm/

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
See PR https://github.com/apache/incubator-nuttx/pull/6197

## Impact
Code refactor

## Testing
Pass CI

### CODE REVIEW

**Review by pussuw — APPROVED** [review]
LGTM

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: libs/libc/machine/arm/aeabi_atexit.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#include <nuttx/compiler.h>
+
+/****************************************************************************
+ * Public Function Prototypes
+ ****************************************************************************/
+
+int __cxa_atexit(CODE void (*func)(FAR void *),
+                 FAR void *object,
+                 FAR void *dso_handle);
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/aeabi_atexit.c:33]:
```suggestion
int __cxa_atexit(void (*func)(FAR void *),
                 void *object,
                 void *dso_handle);
```

File: libs/libc/machine/arm/aeabi_atexit.c [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *   __aeabi_atexit (NULL, f, NULL).  But in the usage model here, static
+ *   constructors are initialized at power up and are never destroyed
+ *   because they have global scope and must persist for as long as the
+ *   embedded device is powered on.
+ *
+ ****************************************************************************/
+
+int weak_function __aeabi_atexit(FAR void *object,
+                                 CODE void (*func)(FAR void *),
+                                 FAR void *dso_handle)
```
Comment (pkarashchenko) [inline libs/libc/machine/arm/aeabi_atexit.c:53]:
```suggestion
int weak_function __aeabi_atexit(void *object,
                                 void (*func)(FAR void *),
                                 void *dso_handle)
```

**Review by pkarashchenko — CHANGES_REQUESTED** [review]

File: libs/libc/machine/arm/aeabi_atexit.c (hunk 2 above)
Comment (xiaoxiang781216) [reply libs/libc/machine/arm/aeabi_atexit.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**